tfhe/src/high_level_api/tag.rs

```diff
@@ -0,0 +1,420 @@
+use crate::high_level_api::backward_compatibility::tag::TagVersions;
+use tfhe_versionable::{Unversionize, UnversionizeError, Versionize, VersionizeOwned};
+
+const STACK_ARRAY_SIZE: usize = std::mem::size_of::<Vec<u8>>() - 1;
+
+/// Simple short optimized vec, where if the data is small enough
+/// (<= std::mem::size_of::<Vec<u8>>() - 1) the data will be stored on the stack
+///
+/// Once a true heap allocated Vec was needed, it won't be deallocated in favor
+/// of stack data.
+#[derive(Clone, Debug)]
+pub(in crate::high_level_api) enum SmallVec {
```

Comment:
does the size of the enum == the size of Vec<u8> ?

Reply:
No the save of the enum is likely slightly bigger as the enum handles the tag.

To have it have the same size would would have to do it ourselves with a union and a u8 tag that serves as both tag/len of data in stack array, but that requires unsafe code so that were it start not to be worth to actuall y do